pkg/controllers/managementuser/rbac/globalrolebinding_handler.go

```diff
@@ -181,38 +186,41 @@ func (c *grbHandler) ensureProvisioningClusterAdminBinding(obj *v3.GlobalRoleBin
 	provCluster := pClusters[0]
 
 	subject := rbac.GetGRBSubject(obj)
-	rbName := name.SafeConcatName(rbac.ProvisioningClusterAdminName(provCluster), rbac.GetGRBTargetKey(obj))
-
-	existingRb, err := c.rbLister.Get(provCluster.Namespace, rbName)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if existingRb != nil {
-		return nil
-	}
+	crtbName := fmt.Sprintf("crtb-%s-%s-%s", rbac.GetGRBTargetKey(obj), "restricted-admin", "cluster-owner")
+	_, err = c.crtbLister.Get(provCluster.Name, crtbName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
 
-	_, err = c.roleBindings.Create(&k8srbac.RoleBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      rbName,
-			Namespace: provCluster.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: obj.APIVersion,
-					Kind:       obj.Kind,
-					Name:       obj.Name,
-					UID:        obj.UID,
+		// Add the restricted admin user as a member of the downstream cluster
+		// by creating a CRTB in the local custer in the namespace named after the downstream cluster.
+		crtb := v3.ClusterRoleTemplateBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      crtbName,
+				Namespace: provCluster.Name,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: obj.APIVersion,
+						Kind:       obj.Kind,
+						Name:       obj.Name,
+						UID:        obj.UID,
+					},
 				},
 			},
-		},
-		RoleRef: k8srbac.RoleRef{
-			APIGroup: k8srbac.GroupName,
-			Kind:     "Role",
-			Name:     rbac.ProvisioningClusterAdminName(provCluster),
-		},
-		Subjects: []k8srbac.Subject{subject},
-	})
-	if err != nil && !apierrors.IsAlreadyExists(err) {
-		return err
+			ClusterName:      provCluster.Name,
+			RoleTemplateName: "cluster-owner",
+		}
+		if subject.Kind == "Group" {
+			crtb.GroupPrincipalName = subject.Name
+		} else if subject.Kind == "User" {
+			crtb.UserPrincipalName = subject.Name
+		}
```

Comment:
The GRB has this information already so you could skip extracting the subject

```suggestion
			ClusterName:      provCluster.Name,
			RoleTemplateName: "cluster-owner",
			UserName: grb.UserName,
			GroupPrincipalName: grb.GroupPrincipalName,
		}
```

Reply:
@maxsokolovsky To answer your previous questions, the required fields are `UserName` or `GroupPrincipalName`. 

This request will fail if both are set. I am unsure if we should check if both fields are not empty since the GRB itself should not have both set. 
@cmurphy Thoughts?